server/portal/apps/projects/management/commands/projects_id.py

```diff
@@ -1,127 +1,126 @@
-                    ''
-                )
-                if '-' not in prj_id:
-                    continue
-                _, prj_id = prj_id.rsplit('-')
-                try:
-                    prj_id = int(prj_id)
-                except ValueError:
-                    # prj_id is not an int
-                    pass
-
-                if prj_id > latest:
-                    latest = prj_id
-            if prj_id == -1:
-                done = True
-            else:
-                offset += 100
-                prjs = Project.listing(
-                    service_account(),
-                    offset=offset,
-                    limit=limit
-                )
-        return latest
-
-    def _get_latest_project_directory(self):  # pylint:disable=no-self-use
-        """Get latest agave project directory."""
-        offset = 0
-        limit = 100
-        _file = BaseFile(
-            service_account(),
-            system=settings.PORTAL_PROJECTS_ROOT_SYSTEM_NAME,
-            path='/'
-        )
-        dirs = _file.children(offset=offset, limit=limit)
-        done = False
-        latest = -1
-        while not done:
-            dir_id = -1
-            for _dir in dirs:
-                if ('-' not in _dir.name or
-                        not _dir.name.startswith(settings.PORTAL_PROJECTS_ID_PREFIX)):
-                    continue
-                _, dir_id = _dir.name.rsplit('-', 1)
-                try:
-                    dir_id = int(dir_id)
-                except ValueError:
-                    # dir_id is not an int
-                    pass
-
-                if dir_id > latest:
-                    latest = dir_id
-
-            if dir_id == -1:
-                done = True
-            else:
-                offset += 100
-                _file.set_children(None)
-                dirs = _file.children(offset=offset, limit=limit)
-        return latest
+        if '-' not in prj_id:
+            continue
+        _, prj_id = prj_id.rsplit('-')
+        prj_id = int(prj_id)
+
+        if prj_id > latest and (max_project_id is None or prj_id < max_project_id):
+            latest = prj_id
+
+    return latest
+
+
+def get_latest_project_directory(max_project_id=None):
+    """Get latest agave project directory.
+
+    :param max_project_id: If provided, then ignore projects ids that are greater than or equal to this value.
+    """
+    latest = -1
+    for f in iterate_listing(service_account(),
+                             system=settings.PORTAL_PROJECTS_ROOT_SYSTEM_NAME,
+                             path='/'):
+        name = f["name"]
+        if '-' not in name or not name.startswith(settings.PORTAL_PROJECTS_ID_PREFIX):
+            continue
+        _, dir_id = name.rsplit('-', 1)
+        dir_id = int(dir_id)
+        if dir_id > latest and (max_project_id is None or dir_id < max_project_id):
+            latest = dir_id
+    return latest
+
+
+class Command(BaseCommand):
+    help = (
```

Comment:
Should we include a usage example somewhere? On first glance I'm not sure it's obvious you should only choose one of either `--update-using-max-value-found` or `--max-project-id`

Reply:
Cool.  Will catch if user tries using --update and --update-using-max-value-found at the same time (not allowed).

And add some docstrings to describe typical uses 👍

Reply:
@rstijerina , addressed in the last 3 commits.  good to merge now I think.